src/logics/ArithLogic.cc

```diff
@@ -351,6 +351,45 @@ namespace {
         return logic.mkPlus(std::move(args)); // TODO: Can we use non-simplifying versions of mkPlus/mkTimes?
     }
 
+    PTRef polyToPTRefSubstitution(ArithLogic & logic, PTRef var, poly_t & poly) {
+        if ((logic.hasUFs() or logic.hasArrays()) and logic.isVar(var)) {
+            if (std::ranges::any_of(poly, [&logic](auto const & term) {
+                    return term.var != PTRef_Undef and not logic.isVar(term.var);
+                })) {
+                return PTRef_Undef;
+            }
+        }
+
+        bool const hasInts = logic.hasIntegers();
+
+        {
+            poly_t polyCp;
+            poly_t & polyRef = [&]() -> auto & {
+                if (not hasInts) { return poly; }
+                polyCp = poly;
+                return polyCp;
+            }();
+            // non-const operations on poly!
+            auto coeff = polyRef.removeVar(var);
+            coeff.negate();
+            if (not coeff.isOne()) {
+                polyRef.divideBy(coeff);
+                assert(hasInts or not logic.yieldsSortInt(var));
+                if (hasInts and logic.yieldsSortInt(var)) {
+                    if (not std::ranges::all_of(polyRef, [](auto const & term) { return term.coeff.isInteger(); })) {
+                        return PTRef_Undef;
+                    }
+                }
+            }
+
+            if (hasInts) { poly = std::move(polyCp); }
```

Comment:
I believe this whole thing is unnecessary, because equalities in integer logic are normalized (see `ArithLogic::mkBinaryEq` -> `return sumToNormalizedEquality(diff);`) Although, this may not be true if UFs or arrays are present (we should have a look at that).

Normalized means that the coefficients of different terms in the polynomial are coprime.
Thus, I believe we can just check if the coefficient of the variable is one or not, and bail out if it is not.
No need to copy the polynomial.